src/fsharp/CompileOptions.fs

```diff
@@ -904,6 +904,8 @@ let SetTargetProfile tcConfigB v =
         | "netcore"  -> PrimaryAssembly.System_Runtime
         // Indicates we assume "netstandard.dll", i.e .NET Standard 2.0 and above
         | "netstandard"  -> PrimaryAssembly.NetStandard
+        // Indicates we assume "System.Private.CoreLib.dll", this happens when we don't use the reference assemblies for netcore.
+        | "netcore_private"  -> PrimaryAssembly.System_Private_CoreLib
```

Comment:
Unfortunately, we need this in order to compile netcore apps that are not using the reference assemblies, but the implementation ones. We only do this in testing.

I really wish we didn't have the notion of a PrimaryAssembly.

Reply:
Should this be summarized further in an up-for-grab issue?

Reply:
It could be. This is just my opinion; there may be history here where we have to care about this.